python/cudf_polars/cudf_polars/callback.py

```diff
@@ -218,14 +219,17 @@ def _callback(
     assert n_rows is None
     if timer is not None:
         assert should_time
+
+    context = IRExecutionContext()
```

Comment:
This is probably the right spot to create the context for a given query, since we have easy access to `config_options`.